packages/contracts-bedrock/src/L2/TeaWAPOracle.sol

```diff
@@ -35,6 +40,51 @@ contract TeaWAPOracle {
     ///         setting the storage value to the fallback.
     uint256 public constant MAX_ORACLE_DOWNTIME = 5 minutes;
 
+    /// @notice SECURITY FIX: Minimum sane price for TEA/ETH ratio to prevent oracle manipulation
+    ///         Set to 100,000 TEA per ETH (100k * 1e18)
+    /// @dev REASON: Prevents gradual manipulation attacks where attacker slowly drains pool over
+    ///      many blocks to manipulate TWAP (not detectable by flash loan protection).
+    ///      Example attack: Drain pool from 2M → 200M TEA/ETH over 100 blocks, oracle would
+    ///      accept 100x price without bounds, causing users to pay 100x gas fees.
+    /// @dev These boundaries are arbitrary and MUST be adjusted by the Tea team based on
+    ///      expected market conditions and price ranges before mainnet deployment
+    uint160 public constant MIN_SANE_PRICE = 100_000e18;
+
+    /// @notice SECURITY FIX: Maximum sane price for TEA/ETH ratio to prevent oracle manipulation
+    ///         Set to 10,000,000 TEA per ETH (10M * 1e18)
+    /// @dev REASON: Prevents gradual manipulation attacks where attacker slowly drains pool over
+    ///      many blocks to manipulate TWAP (not detectable by flash loan protection).
+    ///      Similar missing bounds have caused millions in losses in other DeFi protocols.
+    /// @dev These boundaries are arbitrary and MUST be adjusted by the Tea team based on
+    ///      expected market conditions and price ranges before mainnet deployment
+    uint160 public constant MAX_SANE_PRICE = 10_000_000e18;
+
```

Comment:
I think this is a useful safeguard but the tricky thing here is that it's unclear what max or min sane price are. Let me consider this one for a bit.